core/common/iterator/src/main/java/org/eclipse/rdf4j/common/iteration/DelayedIteration.java

```diff
@@ -16,6 +16,7 @@
  * or where a created iteration consumes scarce resources like JDBC-connections or memory. Subclasses must implement the
  * <var>createIteration</var> method, which is called once when the iteration is first needed.
  */
+@Deprecated(since = "4.1.0", forRemoval = true)
 public abstract class DelayedIteration<E, X extends Exception> extends AbstractCloseableIteration<E, X> {
```

Comment:
You've marked this (and several other abstract iteration classes) as marked for removal. They're in active use though. What are you aiming to replace them by?

Reply:
In 5.0.0 I plan on doing a search replace for `Iteration` with `CloseableIteration` and then removing the deprecation annotations.

Reply:
Maybe it's simpler if I just use `@Deprecated(since = "4.1.0")` instead.

Reply:
There. What do you think @jeenbroekstra ?